github/Requester.py

```diff
@@ -319,7 +320,29 @@ def __createConnection(self):
             kwds["strict"] = True  # Useless in Python3, would generate a deprecation warning
         if atLeastPython26:  # pragma no branch (Branch useful only with Python 2.5)
             kwds["timeout"] = self.__timeout  # Did not exist before Python2.6
-        return self.__connectionClass(self.__hostname, self.__port, **kwds)
+
+        ##
+        ## Connect through a proxy server with authentication, if http_proxy
+        ## set.
+        ## http_proxy: http://user:password@proxy_host:proxy_port
+        ##
+        try:
+           proxy_uri = os.getenv('http_proxy')
+        except:
+           proxy_uri = None
+
+        if (proxy_uri != None):
```

Comment:
[According to the knowledge card on Google](https://www.google.com/search?q=http_proxy), it is standard. Can we get it in all caps? Like this: `HTTP_PROXY`